examples/calibrate_tsqrtP.py

```diff
@@ -0,0 +1,212 @@
+"""
+import pyspecdata as psd
+import matplotlib.pyplot as plt
+import numpy as np
+from itertools import cycle
+
+colorcyc_list = plt.rcParams["axes.prop_cycle"].by_key()["color"]
+color_cycle = cycle(
+    colorcyc_list
+)  # this can be done more than once to spin up multiple lists
+
+V_atten_ratio = 102.35  # attenutation ratio
+skip_plots = 33  # diagnostic -- set this to None, and there will be no plots
+linear_threshold = 100e-6
+
+
+def prog_plen(desired):
+    """function that takes the coefficients of the linear and nonlinear
+    regions and applies the fit respectively to calculate the pulse time that
+    will return the desired beta value
+    """
+
+    def zonefit(desired):
+        if desired > linear_threshold:
+            return np.polyval(c_linear[::-1], desired)
+        else:
+            return np.polyval(c_nonlinear[::-1], desired)
+
+    ret_val = np.vectorize(zonefit)(desired)
+    if ret_val.size > 1:
+        return ret_val
+    else:
+        return ret_val.item()
+
+
+with psd.figlist_var() as fl:
+    for filename, nodename in [
+        ("240805_calib_amp1_pulse_calib.h5", "pulse_calib_1"),  # high power
+        ("240805_calib_amp0p1_a_pulse_calib.h5", "pulse_calib_3"),  # low power
+        ("240805_calib_amp0p2_a_pulse_calib.h5", "pulse_calib_1"),  # low power
+        ("240805_calib_amp0p05_pulse_calib.h5", "pulse_calib_5"),  # low power
+    ]:
+        d = psd.find_file(
+            filename, expno=nodename, exp_type="ODNP_NMR_comp/test_equipment"
+        )
+        amplitude = d.get_prop("acq_params")["amplitude"]
+        fl.basename = f"amplitude = {amplitude}"
+        if not d.get_units("t") == "s":
+            print(
+                "************ AG still needs to finish pyspecdata PR to save units!!! ************"
+            )
+            d.set_units("t", "s")  # why isn't this done already??
+        d *= V_atten_ratio
+        d /= np.sqrt(50)  # V/sqrt(R) = sqrt(P)
+
+        def switch_to_plot(d, j):
+            thislen = d["t_pulse"][j]
+            fl.next(f"pulse length = {thislen}")
+
+        def indiv_plots(d, thislabel, thiscolor):
+            if skip_plots is None:
+                return
+            for j in range(len(d["t_pulse"])):
+                if j % skip_plots == 0:
+                    switch_to_plot(d, j)
+                    fl.plot(
+                        d["t_pulse", j],
+                        alpha=0.2,
+                        color=thiscolor,
+                        label=thislabel,
+                    )
+
+        indiv_plots(d, "raw", "blue")
+        # {{{ data is already analytic, and downsampled to below 24 MHz
+        indiv_plots(abs(d), "analytic", "orange")
+        d.ft("t")
+        # {{{ apply frequency filter
+        d.ift("t")
+        dt = d["t"][1] - d["t"][0]
+        SW = 1 / dt
+        carrier = d.get_prop("acq_params")["carrierFreq_MHz"] * 1e6
+        if int(carrier / SW) % 2 == 0:
+            center = carrier % SW
```

Comment:
the math for this and the next are not quite right.  You are not accounting for the fact that it's shifted, so it runs from -SW/2 to SW/2.  I think you wan to first add SW/2, then modulo with SW, then subtract back out, or something like that, but it should be possible to do this with one expression

Reply:
took inspiration from [here](https://www.ni.com/en/shop/data-acquisition/measurement-fundamentals/analog-fundamentals/acquiring-an-analog-signal--bandwidth--nyquist-sampling-theorem-.html) and added comments

Reply:
now updated pulling equation directly from cavanagh

Reply:
Please don't waste time by ignoring the code and comments that I wrote:

If your carrier is slightly greater than your spectrum width, this will not trigger the if statement. But more than that, this is not taking into account the fact that your FT is shifted. We are not starting from zero. We are starting from SW/2. 

Please roll back to the version that I reviewed, test that code in IPython, and debug that code if need it.

Reply:
rolled back and debugged